### PR TITLE
PsrAutoloadingFixer - Fix PSR autoloading outside configured directory

### PR DESCRIPTION
**Pull request by kelunik**
Fixes https://github.com/amphp/php-cs-fixer-config/issues/5.

### CODE REVIEW

File: src/Fixer/Basic/PsrAutoloadingFixer.php [hunk 1]
```diff
@@ -137,6 +137,10 @@ protected function createConfigurationDefinition()
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
+        if (null !== $this->configuration['dir'] && 0 !== strpos($file->getRealPath(), realpath($this->configuration['dir']))) {
```
Comment (keradus) [inline src/Fixer/Basic/PsrAutoloadingFixer.php:140]: without any test added, it's hard to see what you are fixing.
please, add a test

Comment (kelunik) [reply src/Fixer/Basic/PsrAutoloadingFixer.php:140]: Done.

**Review by kubawerlos — APPROVED** [review]

**Comment by SpacePossum** [comment]
Thanks for the PR as bug fix :+1: 

Maybe we should validate that the configured directory exists or at least check the result of the `realpath` call?
I think we can also call the realpath just one time for each run (so not for each file that is processed).

**Comment by clxmstaab** [comment]
should/will this PR als fix https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5537 ?

**Comment by keradus** [comment]
@clxmstaab no,
reason in https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5537#issuecomment-796668065

**Comment by keradus** [comment]
I let myself to apply suggestion of @SpacePossum

**Comment by keradus** [comment]
Thank you @kelunik.

**Comment by kelunik** [comment]
Thank you, we can finally enable the rule again. :-)

**Pull request merged**